src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +835,402 @@ def __try_add_error(error_list, detail):
+        else:
+            # code == 1 (Error), add everything in the errors['details'] to truncated_detail_list
+            truncated_detail_list.extend(errors_details)
+
+        return self.__recompose_truncated_msg(truncated_status_file, truncated_package_list, truncated_detail_list,
+            count_total_errors, substatus_json, substatus_name, substatus_index)
+
+    def __recompose_truncated_msg(self, truncated_status_file, truncated_package_list, truncated_detail_list, count_total_errors, truncated_substatus_json, substatus_name, substatus_index):
+        """ Recompose status file with new errors detail list, new errors message, and truncated patches  """
+        truncated_substatus_msg = None
+        truncated_errors_json = self.__recompose_substatus_errors(truncated_detail_list, count_total_errors)    # Recompose substatus message errors json
+
+        # Recompose assessment substatus message
+        if substatus_name == Constants.PATCH_ASSESSMENT_SUMMARY:
+            # Add assessment tombstone per classifications into packages_in_assessment
+            truncated_package_list.extend(self.__create_assessment_tombstone_list(self.__assessment_packages_removed))
+            truncated_substatus_msg = self.__recompose_assessment_message(truncated_substatus_json, truncated_package_list, truncated_errors_json)
+
+        # Recompose installation substatus message
+        if substatus_name == Constants.PATCH_INSTALLATION_SUMMARY:
+            # Add installation tombstone record
+            # Todo need further requirements to decompose installation tombstone by classifications
+            truncated_package_list.append(self.__create_installation_tombstone())
+            truncated_substatus_msg = self.__recompose_installation_message(truncated_substatus_json, truncated_package_list, truncated_errors_json)
+
+        truncated_status_file['status']['substatus'][substatus_index]['formattedMessage']['message'] = json.dumps(truncated_substatus_msg)
+
+        return truncated_status_file
+
+    def __recompose_substatus_errors(self, truncation_detail_list, count_total_errors):
+        """ Recompose truncated substatus errors json """
+        error_msg = Constants.StatusTruncationConfig.TRUNCATION_WARNING_MESSAGE
+        truncated_error_detail = self.__set_error_detail(Constants.PatchOperationErrorCodes.TRUNCATION, error_msg)      # Reuse the errors object set up
+        self.__try_add_error(truncation_detail_list, truncated_error_detail)
+        truncated_errors_json = self.__set_errors_json(count_total_errors, truncation_detail_list, True)    # True for truncated
+
+        return truncated_errors_json
+
+    def __recompose_assessment_message(self, assessment_substatus_msg, assessment_packages, assessment_errors):
+        """ Recompose assessment substatus message """
+        activity_id = assessment_substatus_msg['assessmentActivityId']
```

Comment:
This is introducing a dependency on the substatus keys/field name (i.e. assessmentActivityId, rebootPending, startTime, etc). Any changes to these field names in future will need changes in this function and in __compose_assessment_substatus_msg()

Why not reuse __compose_assessment_substatus_msg() passing an empty package list to compute assessment substatus without packages instead?

Reply:
Just saw you are using __compose_assessment_substatus_msg below. 

The concern on field name duplication still exists. Suggestion: we could add all of these field names in an enum in Constants and use the reference from those in here.